x/incentive/keeper/rewards_earn.go

```diff
@@ -78,14 +77,17 @@ func (k Keeper) accumulateEarnBkavaRewards(ctx sdk.Context, rewardPeriod types.M
 	// bkava vault denoms from past incentive indexes, may include vaults
 	// that were fully withdrawn.
 	k.IterateEarnRewardIndexes(ctx, func(vaultDenom string, indexes types.RewardIndexes) (stop bool) {
-		if strings.HasPrefix(vaultDenom, "bkava-") {
+		if k.liquidKeeper.IsDerivativeDenom(ctx, vaultDenom) {
 			bkavaVaultsDenoms[vaultDenom] = true
 		}
 
 		return false
 	})
 
-	totalBkavaSupply := k.liquidKeeper.GetTotalDerivativeSupply(ctx)
+	totalBkavaValue, err := k.liquidKeeper.GetTotalDerivativeValue(ctx)
+	if err != nil {
+		panic(err)
```

Comment:
are panics ok in the BeginBlocker? Or should this just log and return early

Reply:
In other modules we've returned errors, then panicked in abci.go if an error is returned.
I'd probably do that (ie panic rather than log and skip), then if there's problems in testnet we'll know about them.